src/test/java/edu/kpi/testcourse/storage/UrlRepositoryFakeImplTest.java

```diff
@@ -38,5 +39,29 @@ void shouldNotAllowToCreateSameAliases() {
     }).isInstanceOf(UrlRepository.AliasAlreadyExist.class);
   }
 
+  @Test
+  void shouldGetAllAliasesForUser() {
+    //GIVEN
+    UrlRepository repo = new UrlRepositoryFakeImpl();
+
+    //WHEN
+    UrlAlias alias1 = new UrlAlias("http://r.com/short", "http://g.com/long1", "aaa@bbb.com");
+    repo.createUrlAlias(alias1);
+
+    //THEN
+    assertThat(repo.getAllAliasesForUser(alias1.email())).isNotNull();
+  }
 
+  @Test
+  void shouldGetAllAliasesForUser_throwPermissionDenied() {
+    //GIVEN
+    UrlRepository repo = new UrlRepositoryFakeImpl();
+
+    //WHEN
+    UrlAlias alias1 = new UrlAlias("http://r.com/short", "http://g.com/long1", "aaa@bbb.com");
+    repo.createUrlAlias(alias1);
+
+    //THEN
+    assertThrows(PermissionDenied.class, ()->{repo.getAllAliasesForUser("bbb@aaa.com");});
```

Comment:
It's better to check on our Interface "PermissionDenied"
```suggestion
    assertThrows(UrlRepository.PermissionDenied.class, ()->{repo.getAllAliasesForUser("bbb@aaa.com");});
```